### DISCUSSION TITLE
Passing too many or too few arguments to the `Confirm`/`Error`/`Message`/`StrSubstNo` method should raise a warning

### DISCUSSION THREAD

**Discussion by rvanbekkum** [StefanMaron/BusinessCentral.LinterCop#252] [New Rule]
If you have a call to the `Confirm`/`Error`/`Message`/`StrSubstNo` method and use a `Label` with placeholders you have the option to pass additional arguments for the placeholders' values that will be substituted into the placeholder locations.

There is currently no code analyzer (afaik) that raises a warning if there are too many or too few/little arguments passed in a call to the `Confirm` method.

Sample:
```al
    procedure SampleProcedurePlaceholders()
    var
        CheckUpWithUserQst: Label 'Are you doing okay %1?', Comment = '%1 = User Name';
        EmotionLessResponseMsg: Label 'Ok, %1', Comment = '%1 = Feeling';
        MyText: Text;
    begin
        if Confirm(CheckUpWithUserQst) then
            Message(EmotionLessResponseMsg)
        else
            Error(EmotionLessResponseMsg);

        MyText := StrSubstNo(EmotionLessResponseMsg);
    end;
```

No code analyzer rules are triggered (except for [LC0021](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0021)).

**Comment by ChrisBlankDe** [comment]
I think [AA0131](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0131) should handle this.

microsoft [knows/knew](https://github.com/microsoft/AL/issues/6456) about the fact that this does not work properly with the confirm. due to an [incomprehensible way of working](https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=1792705994711040), the issue was closed.
